### DISCUSSION TITLE
Could the YI company force remote firmware upgrades ?

### DISCUSSION THREAD

**Discussion by nemonoid** [roleoroleo/yi-hack-Allwinner-v2#700] [General]
One day YI could decide to block the possibility to apply the yi-hack...  to push people to subscribe to their servers.

So I'm wondering, can they technically force a firmware upgrade from remote ?

I guess without the yi-hack it's possible, because if you can push a firmware upgrade from the app, they can probably do it from remote too. 
But with the yi-hack installed ? Could they still do it ?

( obviously if you "disable cloud" they cannot. But I'm talking if you have yi-hack installed but you leave the cloud enabled. )

**Comment by roleoroleo** [comment]
If the cloud is enabled, it's possible.

**Comment by nemonoid** [comment]
Thank you for the reply,
( and thanks for the amazing work you are doing with this repo! )

Do you have any idea of any hack we could do to stop that possibility ? 

It would be a pain to have all the cameras configured and then one day they all stop working because they got a remotely-pushed firmware upgrade.
Yes, one can disable the cloud features, but a lot of people likes the possibility of using both cloud and the yi-hack features...

**Reply by roleoroleo** [reply]
If you use the cloud, you have a bidirectional channel with the enemy.
And we don't know what the processes that are on board the cam can do.

**Reply by nemonoid** [reply]
True.
